hash.go

```diff
@@ -51,3 +53,33 @@ func Hash(digests [][32]byte, chunks [][32]byte) error {
 func HashChunks(digests [][32]byte, chunks [][32]byte) {
 	_hash(&digests[0][0], chunks, uint32(len(chunks)/2))
 }
+
+func HashByteSlice(digests []byte, chunks []byte) error {
+	if len(chunks) == 0 {
+		return nil
+	}
+	if len(chunks)%64 != 0 {
+		return fmt.Errorf("chunks not multiple of 64 bytes")
+	}
+	if len(digests)%32 != 0 {
+		return fmt.Errorf("digests not multiple of 32 bytes")
+	}
+	if len(digests) < len(chunks)/2 {
+		return fmt.Errorf("not enough digest length, need at least %d, got %d", len(chunks)/2, len(digests))
+	}
+	header := *(*reflect.SliceHeader)(unsafe.Pointer(&chunks))
+	header.Len <<= 5
+	header.Cap <<= 5
```

Comment:
Why do you need to shift these values?

Reply:
len and cap are multiplied by 32 since the slice will be 32*n bytes long

Reply:
OK, makes sense. Thanks

@potuz could you add some commentary here for future readers please?